SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java

```diff
@@ -190,4 +206,88 @@ private void updateLongRunningTasksUsage(Map<ResourceUsageType, Number> longRunn
+      Optional<SingularityDeploy> maybeDeploy = deployManager.getDeploy(utilization.getRequestId(), utilization.getDeployId());
+
+      if (maybeDeploy.isPresent() && maybeDeploy.get().getResources().isPresent()) {
+        boolean includeUtilization = true;
+        long memoryBytesReserved = (long) (maybeDeploy.get().getResources().get().getMemoryMb() * SingularitySlaveUsage.BYTES_PER_MEGABYTE);
+        double cpuReserved = maybeDeploy.get().getResources().get().getCpus();
+
+        double unusedCpu = cpuReserved - utilization.getAvgCpuUsed();
+        long unusedMemBytes = memoryBytesReserved - utilization.getMemBytesTotal();
+
+        if (unusedCpu / cpuReserved >= minUnderUtilizedPct) {
+          numRequestsWithUnderUtilizedCpu++;
+          totalUnderUtilizedCpu += unusedCpu;
+          maxUnderUtilizedCpu = Math.max(unusedCpu, maxUnderUtilizedCpu);
+          minUnderUtilizedCpu = Math.min(unusedCpu, minUnderUtilizedCpu);
+        } else if (unusedCpu < 0) {
+          double overusedCpu = Math.abs(unusedCpu);
+
+          numRequestsWithOverUtilizedCpu++;
+          totalOverUtilizedCpu += overusedCpu;
+          maxOverUtilizedCpu = Math.max(overusedCpu, maxOverUtilizedCpu);
+          minOverUtilizedCpu = Math.min(overusedCpu, minOverUtilizedCpu);
+        } else {
+          includeUtilization = false;
+        }
+
+        if (unusedMemBytes / memoryBytesReserved >= minUnderUtilizedPct) {
+          numRequestsWithUnderUtilizedMemBytes++;
+          totalUnderUtilizedMemBytes += unusedMemBytes;
+          maxUnderUtilizedMemBytes = Math.max(unusedMemBytes, maxUnderUtilizedMemBytes);
+          minUnderUtilizedMemBytes = Math.min(unusedMemBytes, minUnderUtilizedMemBytes);
+        } else if (!includeUtilization) {
+          it.remove();
+        }
+      }
+    }
+
+    double avgUnderUtilizedCpu = totalUnderUtilizedCpu / numRequestsWithUnderUtilizedCpu;
+    double avgOverUtilizedCpu = totalOverUtilizedCpu / numRequestsWithOverUtilizedCpu;
+    double avgUnderUtilizedMemBytes = totalUnderUtilizedMemBytes / numRequestsWithUnderUtilizedMemBytes;
+
+    return new SingularityClusterUtilization(new ArrayList<>(utilizationPerRequestId.values()), numRequestsWithUnderUtilizedCpu, numRequestsWithOverUtilizedCpu,
+        numRequestsWithUnderUtilizedMemBytes, totalUnderUtilizedCpu, totalOverUtilizedCpu, totalUnderUtilizedMemBytes, avgUnderUtilizedCpu,
+        avgOverUtilizedCpu, avgUnderUtilizedMemBytes, maxUnderUtilizedCpu, maxOverUtilizedCpu, maxUnderUtilizedMemBytes, minUnderUtilizedCpu,
+        minOverUtilizedCpu, minUnderUtilizedMemBytes, totalMemBytesUsed, totalMemBytesAvailable, totalCpuUsed, totalCpuAvailable, now);
+  }
+
+
+  @VisibleForTesting
+  void clearOldUsage(List<SingularityTaskUsage> taskUsages, String taskId) {
+    if (taskUsages.size() + 1 > configuration.getNumUsageToKeep()) {
+      long minMillisApart = configuration.getUsageIntervalMultiplier() * configuration.getCheckUsageEveryMillis();
```

Comment:
Yeah, the goal here is to increase the interval between each point (poll frequency * interval) so we have a wider span of data. I thought that would be more telling of usage since a task could be in a bad state (fail, be modified, etc.) for a few consecutive runs. The interval defaults to 3 so that's essentially 45 minutes of data (15 points 3 min apart) rather than 15 minutes (15 points 1 min apart)